source/nRF5xServiceDiscovery.cpp

```diff
@@ -114,38 +123,63 @@ nRF5xServiceDiscovery::setupDiscoveredCharacteristics(const ble_gattc_evt_char_d
 void
 nRF5xServiceDiscovery::progressCharacteristicDiscovery(void)
 {
-    /* Iterate through the previously discovered characteristics cached in characteristics[]. */
-    while ((state == CHARACTERISTIC_DISCOVERY_ACTIVE) && (characteristicIndex < numCharacteristics)) {
+    if(state != CHARACTERISTIC_DISCOVERY_ACTIVE) {
+        return;
+    }
+
+    if(remainingCharacteristic != nRF5xDiscoveredCharacteristic() && numCharacteristics > 0) { 
```

Comment:
it would be nice to enclose multiple boolean sub-expressions into parenthesis. 
`((remainingCharacteristic != nRF5xDiscoveredCharacteristic()) && (numCharacteristics > 0))`